dcpy/connectors/edm/publishing.py

```diff
@@ -1,9 +1,69 @@
-from datetime import date
 from pathlib import Path
+import requests
+import pandas as pd
+from urllib.error import HTTPError
 
 from dcpy.connectors import s3
 
 BUCKET = "edm-publishing"
+BASE_URL = f"https://{BUCKET}.nyc3.digitaloceanspaces.com"
+
+
+## TODO - these functions need some cleaning
+construct_branch_output_data_directory_url = lambda dataset, branch, version: (
```

Comment:
Sounds good - I realize there's also two equivalent ones to go in recipes from `apps/qaqc/src/digital_ocean_utils.py`, I'll move them and do the same. Also I'll add stuff for now in new commits while y'all review rather than rebasing and wiping out stuff, and rebase at the end.

Reply:
Great, thanks. I _think_ there's a way to review changes to commits from rebases, but I never found it. So separate commits sounds nice.